daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/Typing.scala

```diff
@@ -359,7 +359,8 @@ private[validation] object Typing {
             checkExpr(controllers, TParties)
           }
           choiceObservers.foreach {
-            checkExpr(_, TParties) // FIXME #7709, be conditional on: supportsContractObservers
+            introExprVar(param, paramType)
```

Comment:
Good catch!

Reply:
Thanks to Moritz for pointing me in the right direction